### ISSUE TITLE
Error parsing template <path>: 'CommentNode' object is not iterable

### ISSUE THREAD

**Issue by deeprave** [ataylor32/django-friendly-tag-loader#2]
## Environment
Django version = 5.0.6 with django-friendly-tag-loader (thanks to google for finding it, - exactly what I was looking for)

## Description
The above error is triggered when using `{% if_has_tag %}` / `{% endif_has_tag %}`.

## Workaround / fix?
Changing (line 92 in `friendly_loader.py`):
```
    nodelist_true = nodelist_false = CommentNode()
```
to
```
    nodelist_true = nodelist_false = [CommentNode()]
```
corrects the problem and appears to work, both positive and negative cases.

## Caveat
I have not tried this fix with other/earlier versions of Django.

**Comment by ataylor32** [comment]
Will you please provide steps to reproduce this issue? I tried using `{% if_has_tag %}` with Django 5.0.6 and it worked fine for me.

**Comment by deeprave** [comment]
I've reproduced this in a simple case using the following `pip list`, with a neapolitan.CRUDView
```
Package                    Version
-------------------------- -------
asgiref                    3.8.1
django                     5.0.6
django-debug-toolbar       4.3.0
django-filter              24.2
django-friendly-tag-loader 1.3.1
neapolitan                 24.4
sqlparse                   0.5.0
```

I think the key here is that the friendly_load is for a non-existent templatetags lib, here `comments` does not exist:
```
{% extends "base.html" %}
{% load friendly_loader %}
{% friendly_load comments %}

{% block content %}

<h1>{% if object %}Edit {{object_verbose_name}}{% else %}Create {{object_verbose_name}}{% endif %}</h1>

<div>
  <form method="POST" {% if form.is_multipart %}enctype="multipart/form-data" {% endif %}
    action="{% if object %}{{ update_view_url }}{% else %}{{ create_view_url }}{% endif %}" class="dl-form">
    {% csrf_token %}
    {% for field in form %}
    <div class="form-group d-flex">
      <div class="col-1 text-end pe-2">
        <label for="{{ field.id_for_label }}" class="form-label text-end">{{ field.label }}</label>
      </div>
      <div class="col-10">
        {{ field }}
        {% if field.help_text %}
          <small id="{{ field.id_for_label }}_help" class="form-text text-muted">{{ field.help_text }}</small>
        {% endif %}
        {% if field.errors %}
          <ul class="errorlist">
          {% for error in field.errors %}
            <li>{{ error }}</li>
            {% endfor %}
          </ul>
        {% endif %}
        {% if_has_tag render_field_comment %}
          {% render_field_comment for=field.id %}
        {% else %}
          <strong>No commenbts</strong>
        {% endif_has_tag %}
      </div>
    </div>
    {% endfor %}
    </div>
    <div class="row">
      <div class="col-1"></div>
      <div class="col-10">
        <button type="submit"
          class="btn btn-primary rounded-3 px-3 py-2 text-sm fw-bold text-white hover:bg-primary-700 focus:outline-visible">Save</button>
      </div>
  </form>
</div>
{% endblock %}
```
This is a somewhat mangled bootstrap version of the canned neapolitan form template.

And the resulting error in the runserver console:
```
Internal Server Error: /bookmarks/bookmark/new/
Traceback (most recent call last):
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/core/handlers/base.py", line 220, in _get_response
    response = response.render()
               ^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/response.py", line 114, in render
    self.content = self.rendered_content
                   ^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/response.py", line 90, in rendered_content
    template = self.resolve_template(self.template_name)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/response.py", line 72, in resolve_template
    return select_template(template, using=self.using)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/loader.py", line 42, in select_template
    return engine.get_template(template_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/backends/django.py", line 33, in get_template
    return Template(self.engine.get_template(template_name), self)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/engine.py", line 177, in get_template
    template, origin = self.find_template(template_name)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/engine.py", line 159, in find_template
    template = loader.get_template(name, skip=skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/loaders/cached.py", line 57, in get_template
    template = super().get_template(template_name, skip)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/loaders/base.py", line 28, in get_template
    return Template(
           ^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 154, in __init__
    self.nodelist = self.compile_nodelist()
                    ^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 196, in compile_nodelist
    return parser.parse()
           ^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 510, in parse
    raise self.error(token, e)
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 508, in parse
    compiled_result = compile_func(self, token)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/loader_tags.py", line 296, in do_extends
    if nodelist.get_nodes_by_type(ExtendsNode):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 1006, in get_nodes_by_type
    nodes.extend(node.get_nodes_by_type(nodetype))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 990, in get_nodes_by_type
    nodes.extend(nodelist.get_nodes_by_type(nodetype))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 1006, in get_nodes_by_type
    nodes.extend(node.get_nodes_by_type(nodetype))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 990, in get_nodes_by_type
    nodes.extend(nodelist.get_nodes_by_type(nodetype))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 1006, in get_nodes_by_type
    nodes.extend(node.get_nodes_by_type(nodetype))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/base.py", line 988, in get_nodes_by_type
    nodelist = getattr(self, attr, None)
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/defaulttags.py", line 313, in nodelist
    return NodeList(self)
           ^^^^^^^^^^^^^^
  File "<path>/test-site/.venv/lib/python3.12/site-packages/django/template/defaulttags.py", line 309, in __iter__
    yield from nodelist
TypeError: 'CommentNode' object is not iterable
```

**Comment by ataylor32** [comment]
Thanks for the information! I was able to reproduce the problem. It appears that extending a template is all it takes for the bug to occur. Your suggested fix works for this:

```
{% if_has_tag this_tag_does_not_exist %}
  {% this_tag_does_not_exist %}
{% else %}
  Fallback content
{% endif_has_tag %}
```

However, it doesn't work without the `{% else %}` part, like this:

```
{% if_has_tag this_tag_does_not_exist %}
  {% this_tag_does_not_exist %}
{% endif_has_tag %}
```

I haven't had time to look into a solution that works in both cases. Do you perhaps have time to look into it?

**Comment by deeprave** [comment]
Ah, I didn't try without the [%else%}. Will dig a bit more, cheers.

**Comment by deeprave** [comment]
Ok, it really needs to be a NodeList, which provides the render method it is looking for. So, this seems to work for all cases:
```
Index: .venv/lib/python3.12/site-packages/friendlytagloader/templatetags/friendly_loader.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.venv/lib/python3.12/site-packages/friendlytagloader/templatetags/friendly_loader.py b/.venv/lib/python3.12/site-packages/friendlytagloader/templatetags/friendly_loader.py
--- a/.venv/lib/python3.12/site-packages/friendlytagloader/templatetags/friendly_loader.py	
+++ b/.venv/lib/python3.12/site-packages/friendlytagloader/templatetags/friendly_loader.py	(date 1716291022070)
@@ -1,4 +1,4 @@
-from django.template import Library, TemplateSyntaxError
+from django.template import Library, NodeList, TemplateSyntaxError
 
 try:
     from django.template.base import TokenType
@@ -89,7 +89,7 @@
     end_tag = 'end%s' % bits[0]
     has_tag = all([tag in parser.tags for tag in bits[1:]])
     has_tag = (not negate and has_tag) or (negate and not has_tag)
-    nodelist_true = nodelist_false = CommentNode()
+    nodelist_true = nodelist_false = NodeList([CommentNode()])
     if has_tag:
         nodelist_true = parser.parse(('else', end_tag))
         token = parser.next_token()
```

**Comment by deeprave** [comment]
An empty NodeList() works as well, so it is a better solution.

**Comment by ataylor32** [comment]
Thank you! I got the fix implemented. However, I'm trying to switch to a more modern build system and haven't been successful with that yet. If you want to give it a try, feel free. Otherwise, you can wait until I get the time to get it working. Once it's working, I can make a new release.

**Comment by ataylor32** [comment]
I was able to get it working, so this issue is now resolved with version 1.3.2. Thanks for reporting this issue!
